table/table_test.go

```diff
@@ -280,10 +280,10 @@ func (t *TableWritingTestSuite) writeParquet(fio iceio.WriteFileIO, filePath str
 
 func (t *TableWritingTestSuite) createTable(identifier table.Identifier, formatVersion int, spec iceberg.PartitionSpec, sc *iceberg.Schema) *table.Table {
 	meta, err := table.NewMetadata(sc, &spec, table.UnsortedSortOrder,
-		t.getMetadataLoc(), iceberg.Properties{"format-version": strconv.Itoa(formatVersion)})
+		t.location, iceberg.Properties{"format-version": strconv.Itoa(formatVersion)})
 	t.Require().NoError(err)
 
-	return table.New(identifier, meta, t.location, iceio.LocalFS{}, nil)
+	return table.New(identifier, meta, t.getMetadataLoc(), iceio.LocalFS{}, nil)
```

Comment:
`table.New`'s `location` parameter overrides tables metadata location
https://github.com/apache/iceberg-go/blob/5142432b98b90b33c3f6af417de4d40aa3be8946/table/table.go#L311-L315

Reply:
this is slightly confusing since a table can have base location, data location, and metadata location.

Reply:
yea, i should probably rename that parameter.

that `location` parameter is *supposed* to be the metadata location. While the tables base location should come from the Metadata that is passed in.